server/migrate/Schema.java

```diff
@@ -154,6 +157,44 @@ public int hashCode() {
         }
     }
 
+    private class WhenDeclaration {
+        private final Set<Pattern> patterns;
+
+        WhenDeclaration(Pattern when) {
+            patterns = when.getNegationDNF().getPatterns().iterator().next().getPatterns();
```

Comment:
why do we need to do negationDNF?

Reply:
Because the other normal form gave a Set of Statements, but the statements did not distinguish between positive and negated statements.